File: android/app/src/main/java/com/readdle/weather/core/DailyForecast.kt
package com.readdle.weather.core

import android.os.Parcelable
import com.readdle.codegen.anotation.SwiftValue
import kotlinx.parcelize.Parcelize
import java.util.Date

@SwiftValue @Parcelize
data class DailyForecast(
    var date: Date = Date(),
    var minTemp: Float = 0.0f,
    var maxTemp: Float = 0.0f,
    var state: WeatherState = WeatherState.NONE
): Parcelable

File: core/Sources/WeatherCore/Data/DailyForecast.swift
//
// DailyForecast.swift
//

import Foundation

public struct DailyForecast: Codable, Hashable {
    public let date: Date
    public let minTemp: Float
    public let maxTemp: Float
    public let state: WeatherState

    public init(date: Date, minTemp: Float, maxTemp: Float, state: WeatherState) {
        self.date = date
        self.minTemp = minTemp
        self.maxTemp = maxTemp
        self.state = state
    }
}
